### PR TITLE
Add support to reject samples if they fall on compressed chunks.

### PR DESCRIPTION
**Pull request by Harkishen-Singh**
Signed-off-by: Harkishen-Singh <harkishensingh@hotmail.com>

### CODE REVIEW

**Review by antekresic — CHANGES_REQUESTED** [review]
This change disrupts too many function signatures, we should come up with a way to setup this configuration by using different recovery functions. 

Best way would be to compose it so that we call two different `runInserter` functions.

File: pkg/pgmodel/ingestor/ingestor.go
Comment (Harkishen-Singh) [reply pkg/pgmodel/ingestor/ingestor.go:45]: I feel a config struct for just a bool seems odd. Unless this becomes too long, abstracting with a struct may not be needed. WDYT?

File: pkg/pgmodel/ingestor/copier.go
Comment (cevian) [inline pkg/pgmodel/ingestor/copier.go:36]: hmm. Why don't we pass down ingestor.Cfg instead and add this as a field to that struct?

File: pkg/pgmodel/ingestor/copier.go
Comment (cevian) [inline pkg/pgmodel/ingestor/copier.go:172]: hmm feels a bit weird to return the error just to catch it one step up and interpret decompressChunk in multiple places. Why don't we instead change tryRecovery to return (retryInsert bool, err error) and return false, nil here?

File: pkg/pgmodel/ingestor/ingestor.go
Comment (cevian) [inline pkg/pgmodel/ingestor/ingestor.go:46]: This should be IgnoreCompressedSamples bool instead so that the default is the zero-value(false). That way we can avoid DecompressChunks: true all over the place.

**Review by cevian — CHANGES_REQUESTED** [review]

File: pkg/pgmodel/ingestor/ingestor.go
Comment (Harkishen-Singh) [reply pkg/pgmodel/ingestor/ingestor.go:46]: I had thought about that, but felt that decompressChunks is easier to read and suits our case.

File: pkg/pgmodel/ingestor/copier.go
Comment (Harkishen-Singh) [reply pkg/pgmodel/ingestor/copier.go:36]: We already have decompressChunks in the struct. I assume you mean to pass cfg just on this level.

**Comment by antekresic** [comment]
As discussed with @Harkishen-Singh, we are going with a handler approach with will be called when you get a "chunk compressed" error and setting it to a noop handler if you disable decompression on insert.

File: pkg/pgmodel/ingestor/dispatcher.go
Comment (antekresic) [inline pkg/pgmodel/ingestor/dispatcher.go:63]: I think we should add a log here, to let the user know we are not ingesting old data. Which means this handler should be declared elsewhere, not inlined.

**Review by antekresic — APPROVED** [review]
Looks good, one comment to add some logging. After that should be good to merge.

**Review by cevian — COMMENTED** [review]
I'll let Ante review this

**Comment by Harkishen-Singh** [comment]
> I'll let Ante review this

@cevian would you mind cancelling the requested changes request as its blocking from a merge.

File: pkg/tests/end_to_end_tests/insert_compressed_chunks_test.go
Comment (antekresic) [inline pkg/tests/end_to_end_tests/insert_compressed_chunks_test.go:45]: You should run this only when running with `useTimescaleDB` flag.

File: pkg/tests/end_to_end_tests/insert_compressed_chunks_test.go
Comment (antekresic) [inline pkg/tests/end_to_end_tests/insert_compressed_chunks_test.go:74]: Only with `useTimescaleDB`.

**Review by antekresic — CHANGES_REQUESTED** [review]
Logging stuff needs a bit more refactoring.

**Review by antekresic — APPROVED** [review]

**Comment by Harkishen-Singh** [comment]
Do not merge.

**Comment by Harkishen-Singh** [comment]
Now mergeable

**Pull request merged**
